x/emissions/keeper/inference_synthesis/network_regrets.go

```diff
@@ -173,186 +178,199 @@ func GetCalcSetNetworkRegrets(
 			return errorsmod.Wrapf(err, "failed to get inferer regret")
 		}
 		newInfererRegret, err := ComputeAndBuildEMRegret(
-			networkLosses.NaiveValue,                                // L^-_i
+			args.NetworkLosses.NaiveValue,                           // L^-_i
 			networkLossesByWorker.InfererLosses[infererLoss.Worker], // L_ij
 			lastRegret.Value,
-			alpha,
+			args.AlphaRegret,
 			blockHeight,
 		)
 		if err != nil {
 			return errorsmod.Wrapf(err, "Error computing and building inferer regret")
 		}
-		err = k.SetNaiveInfererNetworkRegret(ctx, topicId, infererLoss.Worker, newInfererRegret)
+		err = args.K.SetNaiveInfererNetworkRegret(args.Ctx, args.TopicId, infererLoss.Worker, newInfererRegret)
 		if err != nil {
 			return errorsmod.Wrapf(err, "Error setting inferer regret")
 		}
 	}
 
-	sort.Slice(networkLosses.OneOutInfererValues, func(i, j int) bool {
-		return networkLosses.OneOutInfererValues[i].Worker < networkLosses.OneOutInfererValues[j].Worker
+	sortedOneOutInfererValues := args.NetworkLosses.OneOutInfererValues
+	sort.Slice(sortedOneOutInfererValues, func(i, j int) bool {
+		return sortedOneOutInfererValues[i].Worker < sortedOneOutInfererValues[j].Worker
 	})
 
 	// R^-j′ij - One-out inferer inferer regrets
-	for _, oneOutInfererLoss := range networkLosses.OneOutInfererValues {
-		for _, infererLoss := range networkLosses.InfererValues {
-			lastRegret, _, err := k.GetOneOutInfererInfererNetworkRegret(ctx, topicId, oneOutInfererLoss.Worker, infererLoss.Worker)
+	for _, oneOutInfererLoss := range sortedOneOutInfererValues {
+		for _, infererLoss := range sortedInfererValues {
+			lastRegret, _, err := args.K.GetOneOutInfererInfererNetworkRegret(args.Ctx, args.TopicId, oneOutInfererLoss.Worker, infererLoss.Worker)
 			if err != nil {
 				return errorsmod.Wrapf(err, "Error getting one-out inferer inferer regret")
 			}
 			newOneOutInfererInfererRegret, err := ComputeAndBuildEMRegret(
 				networkLossesByWorker.OneOutInfererLosses[oneOutInfererLoss.Worker], // L^-_j'i
 				networkLossesByWorker.InfererLosses[infererLoss.Worker],             // L_ij
 				lastRegret.Value,
-				alpha,
+				args.AlphaRegret,
 				blockHeight,
 			)
 			if err != nil {
 				return errorsmod.Wrapf(err, "Error computing and building one-out inferer regret")
 			}
-			err = k.SetOneOutInfererInfererNetworkRegret(ctx, topicId, oneOutInfererLoss.Worker, infererLoss.Worker, newOneOutInfererInfererRegret)
+			err = args.K.SetOneOutInfererInfererNetworkRegret(args.Ctx, args.TopicId, oneOutInfererLoss.Worker, infererLoss.Worker, newOneOutInfererInfererRegret)
 			if err != nil {
 				return errorsmod.Wrapf(err, "Error setting one-out inferer inferer regret")
 			}
 		}
 	}
 
+	sortedOneOutInfererForecasterValues := args.NetworkLosses.OneOutInfererForecasterValues
+	sort.Slice(sortedOneOutInfererForecasterValues, func(i, j int) bool {
+		return sortedOneOutInfererForecasterValues[i].Forecaster < sortedOneOutInfererForecasterValues[j].Forecaster
+	})
```

Comment:
is this not already sorted? does it need sorting here?

Reply:
This is already sorted in https://github.com/allora-network/allora-chain/blob/f02b25e839ee32eda8f9c422ea44596c40a6ebe7/x/emissions/keeper/inference_synthesis/network_losses.go#L218

and 

https://github.com/allora-network/allora-chain/blob/f02b25e839ee32eda8f9c422ea44596c40a6ebe7/x/emissions/keeper/inference_synthesis/network_losses.go#L239-L249

will fix now